### PR TITLE
[WFLY-12243]: Monitor resource adapter thread pools.

### PR DESCRIPTION
**Pull request by ehsavoie**
* Adding metrics to `activemq.artemis.client.global.thread.pool` and  `activemq.artemis.client.global.scheduled.thread.pool`.

 Jira: https://issues.jboss.org/browse/WFLY-12243

### CODE REVIEW

**Comment by bstansberry** [comment]
@ehsavoie Checkstyle problem

**Comment by bstansberry** [comment]
@ehsavoie Security manager failures.

**Comment by ehsavoie** [comment]
retest this please

File: messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/MessagingSubsystemRootResourceDefinition.java [hunk 1]
```diff
@@ -19,51 +19,145 @@
...
             .setRequired(false)
             .setAllowExpression(true)
             .setRestartAllServices()
             .build();
 
-    public static final SimpleAttributeDefinition GLOBAL_CLIENT_SCHEDULED_THREAD_POOL_MAX_SIZE = create("global-client-scheduled-thread-pool-max-size", INT)
+    public static final SimpleAttributeDefinition GLOBAL_CLIENT_THREAD_POOL_ACTIVE_COUNT = create(GLOBAL_CLIENT_PREFIX + org.jboss.as.threads.CommonAttributes.ACTIVE_COUNT, INT)
+            .setAttributeGroup("global-client")
+            .setStorageRuntime()
+            .setUndefinedMetricValue(new ModelNode(0))
```
Comment (bstansberry) [inline messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/MessagingSubsystemRootResourceDefinition.java]: Please use ModelNode.ZERO here and for all the others.

File: messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/MessagingSubsystemRootResourceDefinition.java [hunk 2]
```diff
@@ -19,51 +19,145 @@
...
             .setXmlName("thread-pool-max-size")
             .setRequired(false)
             .setAllowExpression(true)
             .setRestartAllServices()
             .build();
 
-    public static final SimpleAttributeDefinition GLOBAL_CLIENT_SCHEDULED_THREAD_POOL_MAX_SIZE = create("global-client-scheduled-thread-pool-max-size", INT)
+    public static final SimpleAttributeDefinition GLOBAL_CLIENT_THREAD_POOL_ACTIVE_COUNT = create(GLOBAL_CLIENT_PREFIX + org.jboss.as.threads.CommonAttributes.ACTIVE_COUNT, INT)
+            .setAttributeGroup("global-client")
+            .setStorageRuntime()
```
Comment (bstansberry) [inline messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/MessagingSubsystemRootResourceDefinition.java]: This isn't necessary, as MRR.registerMetric results in it recording it as runtime regardless of what the AD says.

Comment (bstansberry) [inline messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/MessagingSubsystemRootResourceDefinition.java]: Please add AttributeAccess.FLAG.GAUGE_METRIC or COUNTER_METRIC for each of these.

**Review by bstansberry — CHANGES_REQUESTED** [review]

File: messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/MessagingSubsystemRootResourceDefinition.java (hunk 2 above)
Comment (ehsavoie) [reply messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/MessagingSubsystemRootResourceDefinition.java]: I didn't set it to all metrics as for some I'm not sure if the semantics really match (like LARGEST_THREAD_COUNT and KEEPALIVE_TIME)

Comment (bstansberry) [reply messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/MessagingSubsystemRootResourceDefinition.java]: Per https://prometheus.io/docs/concepts/metric_types/ a LARGEST_THREAD_COUNT sounds like a counter as it counts and is monotonically increasing unless reset to zero.

A KEEPALIVE_TIME doesn't sound like a metric at all; it sounds like a runtime-attribute.

**Review by bstansberry — APPROVED** [review]

**Pull request merged**